include/alpaka/core/Assert.hpp

```diff
@@ -71,32 +61,16 @@ namespace alpaka
 
         namespace detail
         {
-            template<typename TLhs, typename TRhs, typename TSfinae = void>
-            struct AssertGreaterThan;
             template<typename TLhs, typename TRhs>
-            struct AssertGreaterThan<
-                TLhs,
-                TRhs,
-                std::enable_if_t<!std::is_unsigned<TRhs>::value || (TLhs::value != 0u)>>
+            struct AssertGreaterThan
             {
-                ALPAKA_NO_HOST_ACC_WARNING
-                ALPAKA_FN_HOST_ACC static auto assertGreaterThan(TRhs const& lhs) -> void
+                ALPAKA_NO_HOST_ACC_WARNING ALPAKA_FN_HOST_ACC static auto assertGreaterThan(TRhs const& lhs)
                 {
                     alpaka::ignore_unused(lhs);
-                    ALPAKA_ASSERT_OFFLOAD(TLhs::value > lhs);
-                }
-            };
-            template<typename TLhs, typename TRhs>
-            struct AssertGreaterThan<
-                TLhs,
-                TRhs,
-                std::enable_if_t<std::is_unsigned<TRhs>::value && (TLhs::value == 0u)>>
-            {
-                ALPAKA_NO_HOST_ACC_WARNING
-                ALPAKA_FN_HOST_ACC static auto assertGreaterThan(TRhs const& lhs) -> void
-                {
-                    alpaka::ignore_unused(lhs);
-                    // Nothing to do for unsigned types camparing to zero.
+                    if constexpr(std::is_signed_v<TRhs> || (TLhs::value != 0u))
+                        ALPAKA_ASSERT_OFFLOAD(TLhs::value > lhs);
```

Comment:
I think `lhs` should be called `rhs`?

Reply:
You're right. Seems like an error in the original code that I didn't notice during refactoring.

Reply:
Yes, unrelated to your change, but made it unnecessarily difficult to understand this piece.

Reply:
Okay, should be a bit clearer now.